array_methods_assignmen.js

```diff
@@ -0,0 +1,404 @@
+// # Sihame Bazi Array methods lab
+//
+// 1. Given an array of numbers, write a function that
+//returns a new array where every element is
+//incremented by 10.
+//
+// ```
+
+// Input: [1,2,3,4,5]
+// Output: [11,12,13,14,15]
+// ```
+//
+// let array = [1,2,3,4,5];
+// let newArray = array.map((currentElement) =>{
+//    return currentElement + 10
+//  })
+//  console.log(newArray)
+// ============================================
+// 2. Given an array, return a new array where
+//every non-string element is converted
+//to an empty string.
+//
+// ```
+// Input: ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+//
+// let arr2 = ["a", 123, "b", "c", {name: "cat"}];
+// let allStringArr = arr2.map((crrElem, index) => {
+// if (typeof crrElem === 'number' || typeof crrElem === 'object' ) {
+//  return " "
+// } else {
+//  return  crrElem
+// }
+// })
+// console.log(allStringArr)
+
+
+// 3. Write a function that returns only the even elements in an array of numbers.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: [2,4]
+//  let arr3 = [1,2,3,4,5];
+// let filteredArr3 = arr3.filter((currentElement3) => {
+//  if (currentElement3 % 2 ===0) {
+//     return true
+//    } else {
+//     return false
+//     }
+//    })
+//  console.log(filteredArr3)
+
+//===============
+// let array3 =  [1,2,3,4,5]
+// let evenNums = array3.filter((num, index) => {
+//  return num % 2 === 0
+// })
+
+// 4. Write a function that checks if all
+//elements in the array are the same.
+//
+// ```
+// Input: [1,1,1,1,1]
+// Output: true
+// let arr4 = [1,1,1,1,1];
+// let arr4same = arr4.every(elem => {
+//   if (elem === 1 ){
+//     return true
+//   }
+//   return false
+// })
+// console.log(arr4same)
+// //
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+// let arr5 = [1,1,1,1,2]
+// let arr5same = arr5.every(elem => {
+//   if (elem === 1 ){
+//     return true
+//   }
+//   return false
+// })
+// console.log(arr5same)
+// 5. Write a function that returns the sum of
+//every element in an array.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: 15
+//function adder (total,crrElem){
+//   return total + currElem;
+// }
```

Comment:
We didn't use an array method for this question . What array method would make this beneficial?